src/Sulu/Bundle/RouteBundle/Content/Type/PageTreeRouteContentType.php

```diff
@@ -0,0 +1,256 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\RouteBundle\Content\Type;
+
+use Doctrine\ORM\EntityManagerInterface;
+use PHPCR\ItemNotFoundException;
+use PHPCR\NodeInterface;
+use PHPCR\PropertyType;
+use Sulu\Bundle\RouteBundle\Entity\Route;
+use Sulu\Bundle\RouteBundle\Entity\RouteRepositoryInterface;
+use Sulu\Bundle\RouteBundle\Generator\ChainRouteGeneratorInterface;
+use Sulu\Bundle\RouteBundle\Manager\ConflictResolverInterface;
+use Sulu\Component\Content\Compat\PropertyInterface;
+use Sulu\Component\Content\SimpleContentType;
+use Sulu\Component\DocumentManager\DocumentManagerInterface;
+use Sulu\Component\DocumentManager\DocumentRegistry;
+
+/**
+ * Provides page_tree_route content-type.
+ */
+class PageTreeRouteContentType extends SimpleContentType
+{
+    const NAME = 'page_tree_route';
+
+    /**
+     * @var DocumentManagerInterface
+     */
+    private $documentManager;
+
+    /**
+     * @var DocumentRegistry
+     */
+    private $documentRegistry;
+
+    /**
+     * @var ChainRouteGeneratorInterface
+     */
+    private $chainRouteGenerator;
+
+    /**
+     * @var ConflictResolverInterface
+     */
+    private $conflictResolver;
+
+    /**
+     * @var EntityManagerInterface
+     */
+    private $entityManager;
+
+    public function __construct(
+        DocumentManagerInterface $documentManager,
+        DocumentRegistry $documentRegistry,
+        ChainRouteGeneratorInterface $chainRouteGenerator,
+        ConflictResolverInterface $conflictResolver,
+        EntityManagerInterface $entityManager
+    ) {
+        parent::__construct('PageTreeRoute');
+
+        $this->documentManager = $documentManager;
+        $this->documentRegistry = $documentRegistry;
+        $this->chainRouteGenerator = $chainRouteGenerator;
+        $this->conflictResolver = $conflictResolver;
+        $this->entityManager = $entityManager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function read(NodeInterface $node, PropertyInterface $property, $webspaceKey, $languageCode, $segmentKey)
+    {
+        $propertyName = $property->getName();
+
+        /*
+         * TODO
```

Comment:
Is that really a todo, or should we just remove that line? :see_no_evil: @wachterjohannes @alexander-schranz Is that comment ok for you? Does it explain everything well enough?